backend/support/api/admin_api_v1_1_0/create_functions.sql

```diff
@@ -257,25 +257,149 @@ $remove_tribal_access_email$ LANGUAGE plpgsql;
 --         "bob@deathstar.org"
 --     ]
 -- }
-CREATE OR REPLACE FUNCTION admin_api_v1_0_0.remove_tribal_access_emails(params JSON) 
+CREATE OR REPLACE FUNCTION admin_api_v1_1_0.remove_tribal_access_emails(params JSON) 
 returns BOOLEAN
 as $remove_tribal_access_emails$
 DECLARE
     ele TEXT;
     em record;
 BEGIN
-    IF admin_api_v1_0_0_functions.has_admin_data_access('DELETE')
+    IF admin_api_v1_1_0_functions.has_admin_data_access('DELETE')
     THEN 
         FOR em IN (SELECT json_array_elements_text((params->>'emails')::JSON) ele)
         LOOP
-            PERFORM admin_api_v1_0_0.remove_tribal_access_email(json_build_object('email', em.ele)::JSON);
+            PERFORM admin_api_v1_1_0.remove_tribal_access_email(json_build_object('email', em.ele)::JSON);
         END LOOP;
         RETURN 1;
     END IF;
     RETURN 0;
 END;
 $remove_tribal_access_emails$ LANGUAGE plpgsql;
 
+
+
+
+--The function below add_tribal_api_key_access adds read access to a tribal API for a specified email.
+--It checks if the API user has read permissions.
+--If the email already exists in the database, the function returns false.
+--Otherwise, it adds the email with 'read-tribal' permission, logs the event, and returns true.
+
+CREATE OR REPLACE FUNCTION admin_api_v1_1_0.add_tribal_api_key_access(params JSON) 
+RETURNS BOOLEAN
+AS $add_tribal_api_key_access$
+DECLARE 
+    user_exists BOOLEAN;
+BEGIN
+    -- If the API user has read permissions, give it a go
+    IF admin_api_v1_1_0_functions.has_admin_data_access('CREATE') THEN
+    -- Check if the user with the given email
+    SELECT EXISTS (
```

Comment:
Lets indent the body of the if. Because we can.

Reply:
Done. Ready for review.